examples/decoding/plot_receptive_field.py

```diff
@@ -0,0 +1,151 @@
+
+
+# Read in channel positions and create our MNE objects from the raw data
+montage = mne.channels.read_montage('biosemi128')
+montage.selection = montage.selection[:128]
+info = mne.create_info(montage.ch_names[:128], sfreq, 'eeg', montage=montage)
+raw = mne.io.RawArray(raw, info)
+n_channels = len(raw.ch_names)
+
+# Plot a sample of brain and stimulus activity
+fig, ax = plt.subplots()
+lns = ax.plot(scale(raw[:, :800][0].T), color='k', alpha=.1)
+ln1 = ax.plot(scale(speech[0, :800]), color='r', lw=2)
+ax.legend([lns[0], ln1[0]], ['EEG', 'Speech Envelope'], frameon=False)
+ax.set(title="Sample activity", xlabel="Time (s)")
+mne.viz.tight_layout()
+
+###############################################################################
+# Create and fit a receptive field model
+# --------------------------------------
+#
+# We will construct a model to find the linear relationship between the EEG
+# signal and a time-delayed version of the speech envelope. This allows
+# us to make predictions about the response to new stimuli.
+
+# Define the delays that we will use in the receptive field
+tmin, tmax = -.4, .2
+
+# Initialize the model
+rf = ReceptiveField(tmin, tmax, sfreq, feature_names=['envelope'],
+                    estimator=Ridge(alpha=1.), scoring='corrcoef')
+n_delays = int((tmax - tmin) * sfreq) + 2  # +2 to account for 0 + end
+
+n_splits = 3
+cv = KFold(n_splits)
+
+# Prepare model data (make time the first dimension)
+speech = speech.T
+Y, _ = raw[:]  # Outputs for the model
+Y = Y.T
+
+# Iterate through folds, fit the model, and predict/test on held-out data
+coefs = np.zeros((n_splits, n_channels, n_delays))
+scores = np.zeros((n_splits, n_channels))
+for ii, (train, test) in enumerate(cv.split(speech)):
+    print('CV iteration %s' % ii)
+    rf.fit(speech[train], Y[train])
+    scores[ii] = rf.score(speech[test], Y[test])
+    coefs[ii] = rf.coef_[:, 0, :]  # We only have one feature
+times = rf.delays_ / float(rf.sfreq)
+
+# Average scores and coefficients across CV splits
+mean_coefs = coefs.mean(axis=0)
+mean_scores = scores.mean(axis=0)
+
+# Plot mean prediction scores across all channels
+fig, ax = plt.subplots()
+ix_chs = np.arange(n_channels)
+ax.plot(ix_chs, mean_scores)
+ax.axhline(0, ls='--', color='r')
+ax.set(title="Mean prediction score", xlabel="Channel", ylabel="Score ($r$)")
+mne.viz.tight_layout()
+
+###############################################################################
+# Investigate model coefficients
+# ------------------------------
+#
+# Finally, we will look at how the linear coefficients (sometimes
+# referred to as beta values) are distributed across time delays as well as
+# across the scalp. We will recreate `figure 1`_ and `figure 2`_ from [1]_.
+
+# Print mean coefficients across all time delays / channels (see Fig 1 in [1])
+time_plot = -.180  # For highlighting a specific time.
+fig, ax = plt.subplots(figsize=(4, 8))
+vmax = np.abs(mean_coefs.max())
```

Comment:
max_coef = mean_coefs.max() (abs shouldn't be necessary)
ax.pcolormesh ..., vmin=-max_coef, vmax=max_coef)